launchdarkly/client.go

```diff
@@ -0,0 +1,79 @@
+package launchdarkly
+
+import (
+	"net/http"
+	"encoding/json"
+	"bytes"
+	"io/ioutil"
+	"errors"
+	"strconv"
+)
+
+type Client struct {
+	AccessToken string
+}
+
+func (c *Client) GetStatus(path string) (int, error) {
+	expectedStatus := make(map[int]bool)
+	return c.execute("GET", path, nil, expectedStatus, nil)
+}
+
+func (c *Client) Get(path string, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("GET", path, nil, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Post(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("POST", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Patch(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("PATCH", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Delete(path string, expectedStatus map[int]bool) error {
+	_, err := c.execute("DELETE", path, nil, expectedStatus, nil)
+	return err
+}
+
+func (c *Client) execute(method string, path string, body interface{}, expectedStatus map[int]bool, response interface{}) (int, error) {
+	requestBody, err := json.Marshal(body)
+	if err != nil {
+		return 0, err
+	}
+
+	req, err := http.NewRequest(method, "https://app.launchdarkly.com/api/v2"+path, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return 0, err
+	}
+
+	req.Header.Set("Authorization", c.AccessToken)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return 0, err
+	}
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
Typically we close the reader after being done reading from it:

`defer resp.Body.Close()`